vlemish/RemoveElementSolution.cs

```diff
@@ -0,0 +1,36 @@
+public class Solution {
+    public int RemoveElement(int[] nums, int val) {
+        // k holds left part of array not containing `val`
+            int k = nums.Length;
+            for (int i = 0; i < k; i++)
+            {
+                if (nums[i] == val)
+                {
+                    ShiftLeft(nums, i);
+                    k--;
+                    i--;
+                }
+            }
+
+            return k;
+    }
+    
+     public void ShiftLeft(int[] nums, int startIndex)
+        {
+            if (nums.Length - 1 <= startIndex)
```

Comment:
Looks like textbook removal, I thought not caring about elements pass the length or about order can give some more benefits to performance, but good to see it written like this to remember how to shift arrays.